src/libraries/Common/src/Interop/Browser/Interop.HttpHandlerService.cs

```diff
@@ -0,0 +1,535 @@
+                public bool IsBodyUsed => (bool)fetchResponse.GetObjectProperty("bodyUsed");
+                public JSObject Headers => (JSObject)fetchResponse.GetObjectProperty("headers");
+                public JSObject Body => (JSObject)fetchResponse.GetObjectProperty("body");
+
+                public Task<object> ArrayBuffer() => (Task<object>)fetchResponse.Invoke("arrayBuffer");
+                public Task<object> Text() => (Task<object>)fetchResponse.Invoke("text");
+                public Task<object> JSON() => (Task<object>)fetchResponse.Invoke("json");
+
+                public void Dispose()
+                {
+                    // Dispose of unmanaged resources.
+                    Dispose(true);
+                    // Suppress finalization.
+                    GC.SuppressFinalize(this);
+                }
+
+                // Protected implementation of Dispose pattern.
+                protected virtual void Dispose(bool disposing)
+                {
+                    if (disposing)
+                    {
+                        // Free any other managed objects here.
+                        //
+                        abortCts.Cancel();
+
+                        abortRegistration.Dispose();
+                    }
+
+                    // Free any unmanaged objects here.
+                    //
+                    fetchResponse?.Dispose();
+                    abortController?.Dispose();
+                }
+
+            }
+
+            private class BrowserHttpContent : HttpContent
+            {
+                private byte[]? _data;
+                private WasmFetchResponse _status;
+
+                public BrowserHttpContent(WasmFetchResponse status)
+                {
+                    _status = status ?? throw new ArgumentNullException(nameof(status), $"{nameof(status)} cannot be null");
+                }
+
+                private async Task<byte[]> GetResponseData()
+                {
+                    if (_data != null)
+                    {
+                        return _data;
+                    }
+
+                    using (Interop.JavaScript.ArrayBuffer dataBuffer = (Interop.JavaScript.ArrayBuffer)await _status.ArrayBuffer().ConfigureAwait(true))
+                    {
+                        using (Uint8Array dataBinView = new Uint8Array(dataBuffer))
+                        {
+                            _data = dataBinView.ToArray();
+                            _status.Dispose();
+                        }
+                    }
+
+                    return _data;
+                }
+
+                protected override async Task<Stream> CreateContentReadStreamAsync()
+                {
+                    var data = await GetResponseData().ConfigureAwait(true);
+                    return new MemoryStream(data, writable: false);
+                }
+
+                protected override async Task SerializeToStreamAsync(Stream stream, TransportContext? context)
```

Comment:
There are new overloads in .NET 5 that take CancellationToken that should also be overridden.

Reply:
Thanks for validating that.  When looking at the code for the `TransportContext?` signature change I saw some other methods and will be taking a look.

Reply:
Addressed in recent commits.